check_metadata.py

```diff
@@ -72,19 +75,35 @@ def checkFile(filePattern):
     print("")
 
 
-def verifyNoBlacklistWords(fileContents, filelocation):
+def verifyNoDenyListWords(fileContents, filelocation):
     fileContents = fileContents.split('/')
     for word in fileContents:
-        if word in blacklist:
+        if word in denyList:
             print("ERROR -- Found in " + filelocation)
-            sys.exit("ERROR -- " + word + " found, and is on the blacklist.")
-
-def checkStringLength (word):
-    length = len(word)
-    if  length == 40 or length == 20:
-        if warn:
-            return "WARNING -- " + word + " is " + str(length) + " characters long"
+            sys.exit("ERROR -- " + word + " found, and is not allowed.")
 
+def characterScan(fileContents, filelocation):
+    twenty = re.findall("[^A-Z0-9][A][ACGIKNPRS][A-Z]{2}[A-Z0-9]{16}[^A-Z0-9]", fileContents)
+    if (twenty) :
+        for word in twenty:
+            if word[1:-1] in allowList:
+                continue
+            print("ERROR -- " + word[1:-1] + " Found in " + filelocation)
+            # sys.exit("ERROR -- 20 character string found, and might be a secret access key.") 
+    forty = re.findall("[^a-zA-Z0-9_/+=][a-zA-Z0-9_/+=]{40}[^a-zA-Z0-9_/+=]", fileContents)
```

Comment:
The Java parser that scans for 40-char words does not include the underscore char like this regex does. If it did, it wouldn't have misidentified my 40-char string last week. Is this regex up-to-date with what the Java test uses?

Also, the script's allowlist at the end of the file includes 40-char strings that contain underscores. Will the Java parser still mark those strings as problem strings?

Reply:
I am using the regex in the global allowlist, but will check with the build team tomorrow to ensure I have the right pattern.